ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -287,7 +287,7 @@ protected Boolean findNextCompactionAndExecute(boolean collectGenericStats, bool
 
     if (success && compactionService.computeStats()) {
       statsUpdater.gatherStats(ci, conf, CompactorUtil.runJobAsSelf(ci.runAs) ? ci.runAs : table.getOwner(),
-          CompactorUtil.getCompactorJobQueueName(conf, ci, table), msc);
+          CompactorUtil.getCompactorJobQueueName(conf, ci, table), msc, table.getParameters());
```

Comment:
can we make `table.getParameters()` 3rd param?

Reply:
Why 3rd? Why not 1st or 2nd? Could you please share the pros behind this.

Reply:
logical order of significance, we gather stats for a table, so it should be among the first if not the first, not after the supplementary stuff like msc

Reply:
Chenged the parameter order to `conf, ci, tableProperties ...`